### PR TITLE
charts: Fix imagePullSecrets in value schema

### PR DESCRIPTION
**Pull request by knrt10**
k8s wanted it in a different way, earlier we were validating wrong input.

Fixes: #2113

### CODE REVIEW

File: charts/headlamp/values.schema.json [hunk 1]
```diff
@@ -35,7 +35,14 @@
       "type": "array",
       "description": "Image pull secrets",
```
Comment (illume) [inline charts/headlamp/values.schema.json:36]: Would you mind fixing this description too?

Currently someone looking at the description for ImagePullSecrets is told imagePullSecrets is...  "Image pull secrets"

The description from the documentation is good
https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#containers

> ImagePullSecrets is an optional list of references to secrets in the same namespace to use for pulling any of the images used by this PodSpec. If specified, these secrets will be passed to individual puller implementations for them to use. More info: https://kubernetes.io/docs/concepts/containers/images#specifying-imagepullsecrets-on-a-pod

File: charts/headlamp/values.schema.json [hunk 2]
```diff
@@ -35,7 +35,14 @@
...
       "items": {
-        "type": "string"
+        "type": "object",
+        "properties": {
+          "name": {
+            "type": "string",
+            "description": "Name of the image pull secret"
+          }
+        },
+        "required": ["name"]
```
Comment (illume) [inline charts/headlamp/values.schema.json:45]: The docs say the name is not required...

See imagePullSecrets here: https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#containers

If you click [LocalObjectReference](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/local-object-reference/#LocalObjectReference) from there you see name doesn't say it is required. If a field is required, it always says it's required.

**Review by illume — CHANGES_REQUESTED** [review]
👍

File: charts/headlamp/values.schema.json (hunk 1 above)
Comment (knrt10) [reply charts/headlamp/values.schema.json:36]: Thank you, will fix it

File: charts/headlamp/values.schema.json [hunk 3]
```diff
@@ -35,7 +35,13 @@
       "type": "array",
       "description": "Image pull secrets",
       "items": {
-        "type": "string"
+        "type": "object",
+        "properties": {
+          "name": {
+            "type": "string",
+            "description": "ImagePullSecrets is an optional list of references to secrets in the same namespace to use for pulling any of the images used by this PodSpec. If specified, these secrets will be passed to individual puller implementations for them to use."
```
Comment (illume) [inline charts/headlamp/values.schema.json:42]: This is the name field description, but the it reads like it is for the parent. I think this should be moved up.

Comment (knrt10) [reply charts/headlamp/values.schema.json:42]: Oops, my bad. Fixed. Thanks again :-)

**Review by illume — APPROVED** [review]
Thanks 🎉🎈

**Pull request merged**